src/tokens/saga.ts

```diff
@@ -309,6 +333,50 @@ export function* watchAccountFundedOrLiquidated() {
   }
 }
 
+export async function fetchImportedTokenBalances(
+  address: Address,
+  importedTokens: StoredTokenBalances,
+  knownTokenBalances: Record<string, FetchedTokenBalance>
+) {
+  try {
+    const importedTokensList = Object.values(importedTokens)
+    const importedTokensWithBalance: StoredTokenBalances = {}
+
+    const balanceRequests = importedTokensList.map(async (importedToken) => {
+      if (!importedToken) {
+        return
+      }
+
+      let balance
+      if (knownTokenBalances[importedToken.tokenId]) {
+        balance = knownTokenBalances[importedToken.tokenId].balance
+      } else {
+        const contract = getContract({
+          abi: erc20.abi,
+          address: importedToken!.address as Address,
+          client: {
+            public: publicClient[networkIdToNetwork[importedToken.networkId]],
+          },
+        })
+        balance = new BigNumber((await contract.read.balanceOf([address])).toString())
+          .shiftedBy(-importedToken.decimals)
+          .toFixed()
+      }
+
+      importedTokensWithBalance[importedToken.tokenId] = {
+        ...importedToken,
+        balance,
+      }
+    })
+
+    await Promise.all(balanceRequests)
```

Comment:
I tried to execute these requests with a `multicall`, but I don't know what I'm doing wrong, I can't make it work.
I got a `undefined is not a function` when I tried, I can't even get the balance using the `multicall` function.

I think the current implementation works so we can merge it, but I think we should implement the multicall, so maybe I can make a another branch/PR with the code that throws the error and someone can help me find out the error?

Reply:
@jeanregisser maybe you have an idea? 

i also wonder if we should have a try / catch inside each `balanceRequest` promise - it seems like right now, if something fails inside any promise then none of the balances will be updated? (i assume `getContract` and / or `contract.read.balance` can throw, maybe i'm wrong about this?)

Reply:
Branch with an example code: https://github.com/valora-inc/wallet/tree/multicall-error-imported-tokens

Reply:
oh weird, the multicall works on [blockchain-api](https://github.com/valora-inc/blockchain-api/blob/0db7ebaab4cfdf04fc0c1c726926d6c186dd6287/src/blockchain.ts#L25),
maybe and env specific issue with React Native? what's the `undefined` function it complains about?

Reply:
Yeah, I copied the code from there, and it's not working 🙁 
I don't know what's `undefined`, `client.multicall` exists, but when I call it, it throws. So it must be happening inside the `multicall`.

Reply:
i had a quick look out of curiousity. we're on different versions of viem (blockchain-api is still on v1, wallet is on v2). the multicall in the wallet throws at this block https://github.com/wevm/viem/blob/e2ed0690dd6112e3b46779044d85f9c147847d6e/src/actions/public/multicall.ts#L208-L223 complaining that `"aggregate3Results" is read-only`. i don't really know how to interpret this, but this block is a little different to the one in v1 of viem (https://github.com/wevm/viem/blob/d10b8015104169de550f50b4e562e6279e70596e/src/actions/public/multicall.ts#L175-L186)

anyway, i just wanted to note this in case it makes sense to either of you. you can debug the multicall with console statements in `node_modules/viem/_cjs/actions/public/multicall.js` @dievazqu